src/lib/components/AccountInsights.svelte

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+  import { convertValue } from '$lib/conversion'
+  import { BitcoinDenomination } from '$lib/types'
+  import { truncateValue } from '$lib/utils'
+  import { onDestroy, onMount } from 'svelte'
+  import { channelsAPY$ } from '$lib/streams'
+  import { translate } from '$lib/i18n/translations'
+  import type { ChannelAPY } from '$lib/backends'
+  import type { ChartItem } from 'chart.js'
+
+  let showZeroFeesMessage = false
+
+  enum ChartIDs {
+    fees = 'fees',
+    apy = 'apy'
+  }
+  const charts = [
+    {
+      id: ChartIDs.fees,
+      title: $translate('app.labels.routing_chart')
+    },
+    {
+      id: ChartIDs.apy,
+      title: $translate('app.labels.apy_chart')
+    }
+  ]
+
+  async function renderFeesChart(channels: ChannelAPY[]) {
+    const { Chart } = await import('chart.js/auto')
+
+    let labels: string[] = []
+    let data: number[] = []
+
+    channels?.forEach(({ account, fees_in_msat, fees_out_msat }) => {
+      const routingFees = fees_in_msat + fees_out_msat
+
+      if (routingFees) {
+        const value = Math.round(
+          Number(
+            convertValue({
+              value: routingFees.toString(),
+              from: BitcoinDenomination.msats,
+              to: BitcoinDenomination.sats
+            })
+          )
+        )
+
+        labels.push(truncateValue(account, 5))
+        data.push(value)
+      }
+    })
+
+    const el: ChartItem | null = document.getElementById(ChartIDs.fees) as ChartItem
+
+    if (el) {
+      new Chart(el, {
+        type: 'pie',
+        data: {
+          labels,
+          datasets: [
+            {
+              data
+            }
+          ]
+        }
+      })
+    }
+  }
+
+  async function renderApyChart(channels: ChannelAPY[]) {
+    const { Chart } = await import('chart.js/auto')
+
+    let labels: string[] = []
+    let data: number[] = []
+
+    channels?.forEach(({ account, apy_total }) => {
+      const total = Number(apy_total.substring(0, apy_total.length - 1))
+
+      if (total) {
+        labels.push(truncateValue(account, 5))
+        data.push(Number(apy_total.substring(0, apy_total.length - 1)))
+      }
+    })
+
+    const el: ChartItem | null = document.getElementById(ChartIDs.apy) as ChartItem
+
+    if (el) {
+      new Chart(el, {
```

Comment:
Can we capture the return value in a variable so that we can later check if there is a chart and update it if the data updates?

Reply:
We would also need this for any cleanup on destroy as well